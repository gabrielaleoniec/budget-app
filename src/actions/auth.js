import {firebase, googleAuthProvider} from '../firebase/firebase';

export const startLogIn = () => {
    return () => {
        return firebase.auth().signInWithPopup(googleAuthProvider);
    }
}