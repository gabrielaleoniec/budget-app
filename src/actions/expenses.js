import database from '../firebase/firebase';

// ADD EXPENSE
export const addExpense = (expense = {}) => {
    return {
        type: "ADD_EXPENSE",
        expense
    }
}

export const startAddExpense = (expenseData = {}) => {
    return (dispatch) => {
        const {
            description = '',
            note = '',
            amount = 0,
            createdAt = 0
        } = expenseData;

        const date = createdAt ? new Date(createdAt).valueOf() : new Date().valueOf();

        const expense = { description, note, amount, createdAt: date };

        return database.ref('expenses').push(expense).then((ref) => {
            dispatch(addExpense({
                ...expense,
                id: ref.key
            }))
        });
    }
}

// REMOVE EXPENSE
export const removeExpense = ({ id } = {}) => ({
    type: "REMOVE_EXPENSE",
    id: id
});

// START REMOVE EXPENSES
export const startRemoveExpense = ({ id } = {}) => {
    return (dispatch) => {
        return database.ref(`expenses/${id}`).remove().then(() => { 
            dispatch(removeExpense({ id }))
        });
    }
}

// EDIT EXPENSE
export const updateExpense = (id, updates) => ({
    type: "UPDATE_EXPENSE",
    id,
    updates
})

// SET EXPENSES
export const setExpenses = (expenses) => ({
    type: "SET_EXPENSES",
    expenses
});

export const startSetExpenses = () => {
    return (dispatch) => {
        return database.ref('expenses').once('value').then(snapshot => {
            const expenses = [];

            snapshot.forEach(childSnapshot => {
                expenses.push({
                    id: childSnapshot.key,
                    ...childSnapshot.val()
                });
            });
            dispatch(setExpenses(expenses));
        }).catch((e) => {
            console.log('Error', e);
        })
    }
};
