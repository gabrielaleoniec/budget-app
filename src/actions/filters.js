export const setTextFilter = (text = '') => ({
    type: "SET_TEXT_FILTER",
    text
});

export const sortByDate = () => ({
    type: "SORT_BY",
    sortBy: 'createdAt'
});

export const sortByAmount = () => ({
    type: "SORT_BY",
    sortBy: 'amount'
});

export const setStartDate = (startDateString = '') => {
    const startDate = new Date(startDateString);
    const dateAsMiliseconds = Date.parse(startDate);
    return {
        type: "SET_START_DATE",
        startDate: isNaN(dateAsMiliseconds) ? undefined : dateAsMiliseconds
    }
}

export const setEndDate = (endDateString = '') => {
    const endDate = new Date(endDateString);
    const dateAsMiliseconds = Date.parse(endDate);

    return {
        type: "SET_END_DATE",
        endDate: isNaN(dateAsMiliseconds) ? undefined : dateAsMiliseconds
    }
}