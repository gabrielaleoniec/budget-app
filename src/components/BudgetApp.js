import React from 'react';

const BudgetApp = () => {
    return <div>Inside the compppponent</div>
}

export default BudgetApp;