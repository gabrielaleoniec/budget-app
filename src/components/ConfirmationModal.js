import React from 'react';

const ConfirmationModal = (props) => {
    const className = props.showConfirmationBox ? "confirmation-box--show" : '';
    return <div className="box-layout box-layout--cover">
        <div className={`confirmation-box ${className}`}>
            <p>Should the expense be removed?</p>
            <div className="confirmation-box__buttons">
                <button className="button button--weak" onClick={props.hideConfirmationModal}>Cancel</button>
                <button className="button button--remove" onClick={props.removeExpense}>Remove</button>
            </div>
        </div>
    </div>
}

export default ConfirmationModal;
