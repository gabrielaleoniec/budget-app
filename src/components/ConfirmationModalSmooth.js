import React from 'react';

const ConfirmationModalSmooth = () => {
    return () => {
        
    }
}