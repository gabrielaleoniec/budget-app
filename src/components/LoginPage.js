import React from 'react';
import { connect } from 'react-redux';
import { startLogIn } from '../actions/auth'

export const LoginPage = ({startLogIn}) => {
    return (
        <div>
            <h2>Sign in</h2>
            <button onClick={startLogIn}>Login</button>
        </div>);
};

const mapDispatchToProps = (dispatch) => ({
    startLogIn: () => dispatch(startLogIn())
})

export default connect(undefined, mapDispatchToProps)(LoginPage);
