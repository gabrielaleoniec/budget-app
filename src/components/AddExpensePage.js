import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

const AddExpensePage = (props) => (
    <div>
        <h2>Add expense</h2>
        <ExpenseForm onSubmit={(expense) => {
            props.dispatch(addExpense(expense));
            props.history.push('/');
        }}/>
    </div>
);

const mapStateToProps = (state) => ({
    expenses: state.expenses
})

export default connect(mapStateToProps)(AddExpensePage);