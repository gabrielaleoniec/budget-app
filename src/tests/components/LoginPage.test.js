import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';
import { startLogInGoogle, startLogInGitHub } from '../../actions/auth';

test('Should render LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogInGoogle on button click', () => {
    const startLogInGoogle = jest.fn();
    const wrapper = shallow(<LoginPage startLogInGoogle={startLogInGoogle} startLogInGitHub={startLogInGitHub} />);
    wrapper.find('button').first().simulate('click');
    expect(startLogInGoogle).toHaveBeenCalled();
});

test('Should call startLogInGitHub on button click', () => {
    const startLogInGitHub = jest.fn();
    const wrapper = shallow(<LoginPage startLogInGoogle={startLogInGoogle} startLogInGitHub={startLogInGitHub} />);
    wrapper.find('button').last().simulate('click');
    expect(startLogInGitHub).toHaveBeenCalled();
});