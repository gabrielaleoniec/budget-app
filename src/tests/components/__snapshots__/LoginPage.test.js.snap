// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage correctly 1`] = `
<div>
  <h2>
    Sign in
  </h2>
  <button>
    Login
  </button>
</div>
`;
